cgi.go

```diff
@@ -94,53 +62,56 @@ func computeKnownVariables(request *http.Request, p *runtime.Pinner) (cArr [27]C
 
 	ra, raOK := fc.env["REMOTE_ADDR\x00"]
 	if raOK {
-		setKnownServerVariable(p, &cArr, remoteAddr, ra)
+		registerTrustedVar(knownServerKeys["REMOTE_ADDR\x00"], ra, track_vars_array, thread)
 	} else {
-		setKnownServerVariable(p, &cArr, remoteAddr, ip)
+		registerTrustedVar(knownServerKeys["REMOTE_ADDR\x00"], ip, track_vars_array, thread)
 	}
 
 	if rh, ok := fc.env["REMOTE_HOST\x00"]; ok {
-		setKnownServerVariable(p, &cArr, remoteHost, rh) // For speed, remote host lookups disabled
+		registerTrustedVar(knownServerKeys["REMOTE_HOST\x00"], rh, track_vars_array, thread) // For speed, remote host lookups disabled
 	} else {
 		if raOK {
-			setKnownServerVariable(p, &cArr, remoteHost, ip)
+			registerTrustedVar(knownServerKeys["REMOTE_HOST\x00"], ra, track_vars_array, thread)
 		} else {
-			cArr[remoteHost] = cArr[remoteAddr]
+			registerTrustedVar(knownServerKeys["REMOTE_HOST\x00"], ip, track_vars_array, thread)
 		}
 	}
 
-	setKnownServerVariable(p, &cArr, remotePort, port)
-	setKnownServerVariable(p, &cArr, documentRoot, fc.documentRoot)
-	setKnownServerVariable(p, &cArr, pathInfo, fc.pathInfo)
-	setKnownServerVariable(p, &cArr, phpSelf, request.URL.Path)
-	setKnownServerVariable(p, &cArr, documentUri, fc.docURI)
-	setKnownServerVariable(p, &cArr, scriptFilename, fc.scriptFilename)
-	setKnownServerVariable(p, &cArr, scriptName, fc.scriptName)
+	registerTrustedVar(knownServerKeys["REMOTE_PORT\x00"], port, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["DOCUMENT_ROOT\x00"], fc.documentRoot, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["PATH_INFO\x00"], fc.pathInfo, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["PHP_SELF\x00"], request.URL.Path, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["DOCUMENT_URI\x00"], fc.docURI, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["SCRIPT_FILENAME\x00"], fc.scriptFilename, track_vars_array, thread)
+	registerTrustedVar(knownServerKeys["SCRIPT_NAME\x00"], fc.scriptName, track_vars_array, thread)
```

Comment:
While these are user input, they come `request.URL` `sanitizedPathJoin` and are sanitized by `net/http`, so they can't contain any invalid paths or control characters. 

I was thinking about this a lot and tried to write tests, do you maybe know how a 'malicious' path could theoretically look like? It would generally be nice to have tests for these cases.

Reply:
> While these are user input, they come from request.URL and sanitizedPathJoin and are sanitized by net/http, so they can't contain any invalid paths or control characters or be excessively long.

This is assuming FrankenPHP isn't used as a library for something other than Caddy. For example, at work, we use FrankenPHP as a library where the "request" is generated from a job queue and the response is fanned out to other queues—if there is a response.

> I was thinking about this a lot and tried to write tests, do you maybe know how a 'malicious' path could theoretically look like? It would generally be nice to have tests for these cases.

I will try to come up with something.

Reply:
> This is assuming FrankenPHP isn't used as a library for something other than Caddy. For example, at work, we use FrankenPHP as a library where the "request" is generated from a job queue and the response is fanned out to other queues—if there is a response.

Interesting, you'd still have to pass a `*http.Request` though. I'll try and create a test set to see if I can create different output with randomly set paths. But I'm pretty sure PHP variable registration is more about safely registering keys and much less so about values.

Reply:
Alternatively, we may document that the path must be sanitized Go-side in the doc block.